### PR TITLE
v1.7.0

### PR DESCRIPTION
**Pull request by Anush008**
## Tasks for feats introduced in 1.7.0.

- [x] `createShardKeyAsync`
- [x] `deleteShardKeyAsync`
- [x] `discoverAsync`
- [x] `discoverBatchAsync`
- [x] Factory methods for ShardKey, ShardKeySelector TargetExample, Vector
- [x] Unit Tests

### CODE REVIEW

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 1]
```diff
@@ -665,6 +674,76 @@ public ListenableFuture<List<AliasDescription>> listAliasesAsync(@Nullable Durat
...
+	 * Creates a shard key for a collection.
+	 *
+	 * @param createShardKey The request object for the operation.
+	 * @param timeout The timeout for the call.
+	 * @return a new instance of {@link ListenableFuture}
+	 */
+	public ListenableFuture<CreateShardKeyResponse> createShardKeyAsync(CreateShardKeyRequest createShardKey, @Nullable Duration timeout) {
+		String collectionName = createShardKey.getCollectionName();
+		Preconditions.checkArgument(!collectionName.isEmpty(), "Collection name must not be empty");
+		logger.debug("Create shard key'{}'", collectionName);
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:699]: log both shard key and collection name

```suggestion
		logger.debug("Create shard key '{}' for '{}'", shardKey, collectionName);
```

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 2]
```diff
@@ -665,6 +674,76 @@ public ListenableFuture<List<AliasDescription>> listAliasesAsync(@Nullable Durat
 
 	//endregion
 
+	//region ShardKey Management
+
+		/**
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:679]:
```suggestion
	/**
```

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 3]
```diff
@@ -665,6 +674,76 @@ public ListenableFuture<List<AliasDescription>> listAliasesAsync(@Nullable Durat
...
+	 * Deletes a shard key for a collection.
+	 *
+	 * @param deleteShardKey The request object for the operation.
+	 * @param timeout The timeout for the call.
+	 * @return a new instance of {@link ListenableFuture}
+	 */
+	public ListenableFuture<DeleteShardKeyResponse> deleteShardKeyAsync(DeleteShardKeyRequest deleteShardKey, @Nullable Duration timeout) {
+		String collectionName = deleteShardKey.getCollectionName();
+		Preconditions.checkArgument(!collectionName.isEmpty(), "Collection name must not be empty");
+		logger.debug("Delete shard key'{}'", collectionName);
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:732]: log both shard key and collection name

```suggestion
		logger.debug("Delete shard key '{}' for '{}'", shardKey, collectionName);
```

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 4]
```diff
@@ -665,6 +674,76 @@ public ListenableFuture<List<AliasDescription>> listAliasesAsync(@Nullable Durat
...
+
+		ListenableFuture<DeleteShardKeyResponse> future = getCollections(timeout).deleteShardKey(deleteShardKey);
+		addLogFailureCallback(future, "Delete shard key");
+		return Futures.transform(future, response -> {
+			if (!response.getResult()) {
+				logger.error("Shard key could not be deleted for '{}'", collectionName);
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:738]: log both shard key and collection name

```suggestion
				logger.error("Shard key '{}' could not be deleted for '{}'", shardKey, collectionName);
```

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 5]
```diff
@@ -665,6 +674,76 @@ public ListenableFuture<List<AliasDescription>> listAliasesAsync(@Nullable Durat
...
+				throw new QdrantException("Shard key could not be deleted for '" + collectionName);
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:739]: include shard key in exception message

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 6]
```diff
@@ -665,6 +674,76 @@ public ListenableFuture<List<AliasDescription>> listAliasesAsync(@Nullable Durat
...
+		String collectionName = createShardKey.getCollectionName();
+		Preconditions.checkArgument(!collectionName.isEmpty(), "Collection name must not be empty");
+		logger.debug("Create shard key'{}'", collectionName);
+
+		ListenableFuture<CreateShardKeyResponse> future = getCollections(timeout).createShardKey(createShardKey);
+		addLogFailureCallback(future, "Create shard key");
+		return Futures.transform(future, response -> {
+			if (!response.getResult()) {
+				logger.error("Shard key could not be created for '{}'", collectionName);
+				throw new QdrantException("Shard key could not be created for '" + collectionName);
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:706]: Include shard key in log message and exception message.

**Review by russcam — APPROVED** [review]
LGTM, left some suggestions

File: src/main/java/io/qdrant/client/QdrantClient.java (hunk 5 above)
Comment (Anush008) [reply src/main/java/io/qdrant/client/QdrantClient.java]: https://github.com/qdrant/java-client/pull/13/commits/a7da5ed23d513f43f4e262b446bd507a2a19339c

File: src/main/java/io/qdrant/client/QdrantClient.java (hunk 6 above)
Comment (Anush008) [reply src/main/java/io/qdrant/client/QdrantClient.java]: https://github.com/qdrant/java-client/pull/13/commits/a7da5ed23d513f43f4e262b446bd507a2a19339c

**Pull request merged**